homeassistant/components/unifi/button.py

```diff
@@ -56,6 +62,21 @@ async def async_power_cycle_port_control_fn(
     await api.request(DevicePowerCyclePortRequest.create(mac, int(index)))
 
 
+def generate_password(length: int = 18) -> str:
+    """Generate password."""
+    alphabet = string.ascii_letters + string.digits
+    password = "".join(secrets.choice(alphabet) for _ in range(length))
+    return password
+
+
+async def async_change_password_control_fn(
+    api: aiounifi.Controller, obj_id: str
+) -> None:
+    """Change WLAN Password."""
+    new_password = generate_password()
+    await api.request(WlanChangePasswordRequest.create(obj_id, new_password))
```

Comment:
```suggestion
    await api.request(WlanChangePasswordRequest.create(obj_id, secrets.token_urlsafe(32)))
```
wouldn't something like this be enough?

Reply:
Sure, I didn't see this suggestion in the stack overflow link you provided, I went for the accepted answer. xD
But 32? Isn't it too much?

Reply:
I just copy pasted an example on the phone, you seem already to have read more about it so whatever you see fit and usable

Reply:
Gotcha!